extensions/wikia/AdEngine/js/utils/btfBlocker.js

```diff
@@ -0,0 +1,93 @@
+/*global define*/
+define('ext.wikia.adEngine.utils.btfBlocker', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.slotTweaker',
+	'wikia.lazyqueue',
+	'wikia.log',
+	'wikia.window'
+], function (adContext, slotTweaker, lazyQueue, log, win) {
+	'use strict';
+
+	var btfQueue = [],
+		btfQueueStarted = false,
+		logGroup = 'ext.wikia.adEngine.utils.btfBlocker',
+		pendingAtfSlots = []; // ATF slots pending for response
+
+	win.ads = win.ads || {};
+	win.ads.runtime = win.ads.runtime || {};
+	win.ads.runtime.disableBtf = false;
+
+	function processBtfSlot(slot, fillInSlot) {
```

Comment:
What do you think about defining this method inside `decorate` and stop passing `fillInSlot` though all methods (`startBtfQueue`, `onSlotResponse`)?